_posts/2021-03-29-changelog-144.markdown

```diff
@@ -0,0 +1,84 @@
+---
+layout: post
+title: "IntelliJ Rust Changelog #144"
+date: 2021-03-29 17:00:00 +0300
+---
+
+## New Features
+
+<!-- https://github.com/intellij-rust/intellij-rust/pull/6799 -->
+* Complete dependencies by subwords in `Cargo.toml` when using experimental local crates index. This feature is disabled by default
+  for now. To use it, enable `org.rust.crates.local.index` option in `Experimental Feature` dialog (`Help | Find Action` and type `Experimental features` to open the dialog)
+
+  {% include gif-img.html path="/assets/posts/changelog-144/6799" w="600px" %}
+
+<!-- https://github.com/intellij-rust/intellij-rust/pull/6947 -->
+* Support import aliases in `Move` refactoring (`Refactor | Move` or <kbd>F6</kbd>)
+
+  {% include gif-img.html path="/assets/posts/changelog-144/6947" w="600px" %}
+
+<!-- https://github.com/intellij-rust/intellij-rust/pull/6893 -->
+* Import unresolved types from a function signature after performing `Extract Function` refactoring (`Refactor | Extract Method` or
+  <kbd>Ctrl</kbd>+<kbd>Alt</kbd>+<kbd>M</kbd>)
+  
+* [Const generics](https://rust-lang.github.io/rfcs/2000-const-generics.html) support improvements:
+  <!-- https://github.com/intellij-rust/intellij-rust/pull/7001 -->
+  * Support const generics in `Wrong generic arguments number` ([E0107](https://doc.rust-lang.org/error-index.html#E0107)) inspection
+
+  <!-- https://github.com/intellij-rust/intellij-rust/pull/6959 -->
+  * Support `Move Element Right/Left` actions for const generic parameters
+
+  <!-- https://github.com/intellij-rust/intellij-rust/pull/6958 -->
+  * Properly render const generics in signature of a top-level item
+  
+  <!-- https://github.com/intellij-rust/intellij-rust/pull/6948 -->
+  * Fix type rendering with const generic parameters
+
+<!-- https://github.com/intellij-rust/intellij-rust/pull/6966 -->
+* Automatically select the default project Rust toolchain from `Preferences | Languages & Frameworks | Rust` when creating a new project
+
+<!-- https://github.com/intellij-rust/intellij-rust/pull/6066 -->
+*  Use [`todo!()`](https://doc.rust-lang.org/std/macro.todo.html) instead of [`unimplemented!()`](https://doc.rust-lang.org/std/macro.unimplemented.html) in function bodies generated by `Implement members`, `Create function`, and `Replace .unwrap() with match` actions (by [@KarelPeeters])
```

Comment:
```suggestion
*  Use [`todo!()`](https://doc.rust-lang.org/std/macro.todo.html) instead of [`unimplemented!()`](https://doc.rust-lang.org/std/macro.unimplemented.html) in function bodies by the plugin's code generation features like  `Implement members` and `Create function` actions (by [@KarelPeeters])
```

WDYT?

Reply:
Looks good! Maybe mention https://plugins.jetbrains.com/plugin/8182-rust/docs/rust-code-generation.html doc page as well?

Reply:
Good idea!

Reply:
BTW I'm not sure that "plugin" is necessary here. For me, "by [code generation](https://plugins.jetbrains.com/plugin/8182-rust/docs/rust-code-generation.html) features" is clear enough

Reply:
It's up to you here